src/transactions/types.ts

```diff
@@ -257,40 +260,40 @@ export type SimulateTransactionOptions = {
 /**
  * Interface that holds the user data input when generating a single signer transaction
  */
-export interface GenerateSingleSignerRawTransactionInput {
+export interface InputGenerateSingleSignerRawTransactionData {
   sender: HexInput;
   feePayerAddress?: undefined;
   secondarySignerAddresses?: undefined;
-  options?: GenerateTransactionOptions;
-  data: GenerateTransactionPayloadData;
+  options?: InputGenerateTransactionOptions;
+  data: InputGenerateTransactionPayloadData;
 }
 
 /**
  * Interface that holds the user data input when generating a fee payer transaction
  */
-export interface GenerateFeePayerRawTransactionInput {
+export interface InputGenerateFeePayerRawTransactionData {
   sender: HexInput;
   feePayerAddress: HexInput;
   secondarySignerAddresses?: HexInput[];
-  options?: GenerateTransactionOptions;
-  data: GenerateTransactionPayloadData;
+  options?: InputGenerateTransactionOptions;
+  data: InputGenerateTransactionPayloadData;
 }
 
 /**
  * Interface that holds the user data input when generating a multi-agent transaction
  */
-export interface GenerateMultiAgentRawTransactionInput {
+export interface InputGenerateMultiAgentRawTransactionData {
   sender: HexInput;
   secondarySignerAddresses: HexInput[];
   feePayerAddress?: undefined;
-  options?: GenerateTransactionOptions;
-  data: GenerateTransactionPayloadData;
+  options?: InputGenerateTransactionOptions;
+  data: InputGenerateTransactionPayloadData;
 }
 
 /**
  * Unified type that holds all the user data input interfaces when generating different transaction types
  */
-export type GenerateTransactionInput =
-  | GenerateMultiAgentRawTransactionInput
-  | GenerateFeePayerRawTransactionInput
-  | GenerateSingleSignerRawTransactionInput;
+export type InputGenerateTransactionData =
+  | InputGenerateMultiAgentRawTransactionData
```

Comment:
Why is it `Input` and `Data`?

Reply:
I was trying to make it explicit that the input isnt a `MultiAgentRawTransaction` somehow...I wasn't sure how to go about it without resorting to `InputDataForGeneratingAMultiAgentRawTransaction`:(

In general I think all of these need to be renamed but yea we can just use this for now